### DISCUSSION TITLE
Supabase and Langchain - Advanced metadata filter using in operator

### DISCUSSION THREAD

**Discussion by zpydee** [supabase/supabase#16513] [Questions] [answered]
I'm struggling a bit with the advanced metadata filter using langchain's SupabaseVectorStore .

Here's a code snippet...

```
 const vectorStore = await SupabaseVectorStore.fromExistingIndex(
    new OpenAIEmbeddings(),
    {
      client,
      tableName: 'documents',
      queryName: 'match_documents',
    }
  );

const { data: conversation } = await client
    .from('conversations')
    .select('id,uploads(id)')
    .eq('id', conversation_id)
    .single();

  const upload_ids = conversation?.uploads.map((upload: Upload) => upload.id);

  const filter: SupabaseFilterRPCCall = (rpc) =>
    rpc.filter('metadata->>upload_id', 'in', upload_ids);

  const documents = await vectorStore.similaritySearch(prompt, 5, filter);

  console.log('doc count: ', documents.length);
```

for which is get the following error: 

```
Error: Error searching for documents: PGRST100 "failed to parse filter (in.3a8794e4-938a-447a-87f7-292e4a7d3f51)" (line 1, column 4) unexpected "3" expecting "("
    at SupabaseVectorStore.similaritySearchVectorWithScore (webpack-internal:///(sc_server)/../../node_modules/langchain/dist/vectorstores/supabase.js:105:19)
```
is the "in" operator not suported, and if not how can i filter for a given array of "upload_id" ?

any help would be much appreciated

**Comment by GaryAustin1** [comment, marked as answer]
Looking at:      
![image](https://github.com/supabase/supabase/assets/54564956/dce9de3c-400a-4535-8f74-5df16a563a67)
It appears you are supposed to use Supabase filters.  
Your .filter() format is not correct.
One option might be `.in('metadata->>upload_id',upload_ids)` .
The other would be to format .filter() to match this : 
![image](https://github.com/supabase/supabase/assets/54564956/6a6ebb85-a40d-44d7-946e-e202bf4303dc)
Note the formatting of the array as a Postgres list.

**Comment by zpydee** [comment]
Thank you!!!

I'm kinda new to Postgres, so appreciate the explanation (and tips).
